src/coprocessor/dag/expr/builtin_encryption.rs

```diff
@@ -175,6 +209,122 @@ impl ScalarFunc {
         }
         Ok(Some(i64::from(LittleEndian::read_u32(&input[0..4]))))
     }
+
+    pub fn aes_encrypt<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &[Datum],
+    ) -> Result<Option<Cow<'a, [u8]>>> {
+        let crypt = try_opt!(self.children[0].eval_string(ctx, row));
+        let key = try_opt!(self.children[1].eval_string(ctx, row));
+        if ctx.aes_mode.unwrap().iv_required && self.children.len() == 3 {
```

Comment:
Different argument length should be pushed down as different function signatures and it should be fine to just leave it.

Reply:
@zz-jason Please help clearify this point.

Reply:
> Different argument length should be pushed down as different function signatures

Yes, it is. 😄

Reply:
It has been modified